alivee/utils/requests/request_meta.py

```diff
@@ -0,0 +1,72 @@
+import urllib
+
+from utils.requests.exceptions import RequestMethodNotSupportedException
+
+
+class RequestMeta(object):
+    '''
+    Request metadata class which holds:
+        host name, url path, url params, headers, reqeuest body, request method and reqeust protocol.
+    '''
+
+    def __init__(self,
+                 host,
+                 path='',
+                 params=None,
+                 method=None,
+                 headers=None,
+                 body=None,
+                 protocol='https'):
+        if method not in self.validMethods():
+            raise RequestMethodNotSupportedException(
+                "invalid_method={0} for url={1} valid_methods={2}".format(
```

Comment:
nit: Or you can give `RequestMethodNotSupportedException` params like:
class RequestMethodNotSupportedException(Exception):
    def __init__(self, method, url):
        self.method = method
        self.url = url
        self.msg = ....